concent_api/utils/decorators.py

```diff
@@ -0,0 +1,155 @@
+from functools                      import wraps
+
+from django.db                      import transaction
+from django.http                    import JsonResponse
+from django.http                    import HttpResponse
+from django.http                    import HttpResponseNotAllowed
+from django.conf                    import settings
+
+from golem_messages                 import dump
+from golem_messages                 import message
+from golem_messages.exceptions      import FieldError
+from golem_messages.exceptions      import MessageError
+from golem_messages.exceptions      import MessageFromFutureError
+from golem_messages.exceptions      import MessageTooOldError
+from golem_messages.exceptions      import TimestampError
+
+from core.validation                import validate_golem_message_signed_with_key
+from core.exceptions                import Http400
+
+from utils.helpers                  import get_validated_client_public_key_from_client_message
+from utils.shortcuts                import load_without_public_key
+
+from utils                          import logging
+
+
+def require_golem_auth_message(view):
```

Comment:
`require_golem_auth_message` and `require_golem_message` look almost identical (there is some code duplication), could the common code be extracted to some function/additional wrapper?

Reply:
I was thinking about this, but please note that the different code is inside the decorator, wrapped by other code. The solution I am thinking of is to create two functions from code that differs inside, but then those functions have to be passed when decorator is used, in view. This is something I would like to avoid. Maybe you have some idea how to do it better ?